service/interpreter/cadence/worker.go

```diff
@@ -19,6 +19,10 @@ type InterpreterWorker struct {
 	tasklist  string
 }
 
+type StartOptions struct {
+	DisableStickyCache bool
```

Comment:
I know we have a specific use case for this start option to disable sticky cache, but I'm wondering if we should add a warning comment about the performance impacts of doing this for other reasons. (My understanding is that this would have a  not insignificant impact on performance in a non-test/real-world use case. Is that right?)